src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java

```diff
@@ -139,10 +140,10 @@ public void doProcessDownloadRequest(
         }
 
         for (AzureBlob blob : individualBlobs) {
-            if (blob.getBlobName().equals(blobName)) {
+            if (blobName.equals(URLDecoder.decode(blob.getBlobName(), "UTF-8"))) {
```

Comment:
Also archived file should have similar logic here, please consider it too.

Reply:
Hey @gavinfish, 
any recommendations on how to proceed?

Reply:
@Johann01 sorry for the late response here since I was on vocation. You just need to add the parameter like `includeArchiveZips: true` in your downloading command and `uploadZips: true` in your uploading command.

Reply:
Hey @gavinfish, 
long time no see! 
You were right with the assumption that it is the same with archived files as you can see here. 

![image](https://user-images.githubusercontent.com/37617663/53808930-d206b080-3f53-11e9-9e12-3e3e8ef3c533.png)

So I adjusted the zipArchiveBlob logic the same way as the normal blob logic. 
I am gonna create a new pull request for that. 
I also imported the StandardCharsets as suggested by you.